### PR TITLE
feat(config): persist credentials to file

### LINKED ISSUE #96: Persist Astarte credentials secret after register
If no credential secret is present, the Astarte message hub register a new device, this operation return a credentials secret should be preserved even after restart or reboot.

https://github.com/astarte-platform/astarte-message-hub/blob/59d4b0aecc724a608c5a4bcf2846b7d97f8348f1/src/main.rs#L42

### PR DESCRIPTION
**Pull request by joshuachp**
Add a method to obtain the credentials if not present in the configurations. The credentials are stored in the current working directory or in a directory passed from the command line.

If the credentials are not present, it will try to register the device on Astarte with the pairing token.

Added `tempfile` crate to test using the file system.

Closes #96 and #103

### CODE REVIEW

**Comment by joshuachp** [comment]
The credentials and store could also be put in the `config.toml`.

**Comment by joshuachp** [comment]
This closes also #103

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/config/mod.rs [hunk 1]
```diff
@@ -108,6 +126,85 @@ impl MessageHubOptions {
             && valid_interface_dir
             && (!self.pairing_url.is_empty())
     }
+
+    /// Obtains the credential secret from the stored path or by registering the device.
+    pub async fn obtain_credential_secret<'a: 'b, 'b>(
+        &'a mut self,
+    ) -> Result<&'b str, AstarteMessageHubError> {
```
Comment (harlem88) [inline src/config/mod.rs:133]: Why did you apply this lifetime? Wasn't it easier to return a `String`?

**Review by harlem88 — APPROVED** [review]

**Review by sorru94 — CHANGES_REQUESTED** [review]

File: src/config/mod.rs [hunk 2]
```diff
@@ -108,6 +126,85 @@ impl MessageHubOptions {
...
+                    AstarteMessageHubError::FatalError(format!(
+                        "failed to write credentials to {}: {}",
+                        path.to_string_lossy(),
+                        err
+                    ))
+                })?;
+
+                cred
+            }
+        };
```
[the remark this answers is not in the document]
Comment (joshuachp) [reply src/config/mod.rs]: Refactored in **8ee5a875b5259483f36a25ec32993da0502124ac**

**Review by harlem88 — APPROVED** [review]

**Pull request merged**